18_regexp_lab/01.py

```diff
@@ -1,12 +1,28 @@
-"""
-Write a program that reads data
-from property files.
-Each line in the file can either be:
-    An empty line
-    A comment line (Start with #)
-    A property line (of the form key = value)
-
-Write a program that takes a property file name and key
-as command line arguments and prints the requested value
-"""
-
+"""
+Elad Agam - Regex
+01 - print a value of configuration in configuration file 
+"""
+import re
+import argparse
+import os
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-c', '--configuration')
+parser.add_argument('-k', '--key')
+
+args = vars(parser.parse_args())
+
+workingDir =  os.getcwd()
+
+#Read inputs and output files - full path 
+conf = os.path.join(workingDir, args['configuration'])
+key = args['key']
+
+
+with open(conf,"r") as fin:
+    for line in fin:
+        res = re.search(key+r'.*=.*(\d)+', line)
+        if res is not None:
+            print res.group(1)
+
+

```

Comment:
It is better to first read and parse the whole file into a dictionary and then look up the key.
Besides this will match the commented out`# answer = 42` when searching for *answer* but won't match `name = Foo Bar` when searching for *name*.